Math/NumberTheory/GCD.hs

```diff
@@ -252,6 +260,25 @@ cw16 (W16# x#) (W16# y#) = coprimeWord# x# y#
 cw32 :: Word32 -> Word32 -> Bool
 cw32 (W32# x#) (W32# y#) = coprimeWord# x# y#
 
+
+newtype Coprimes a b = Coprimes { getValues :: Map.Map a b } deriving (Eq)
```

Comment:
I'd rather do not use such generic name as `getValues`. I usually follow `newtype Foo a = Foo { unFoo :: a }` convention, so I suggest to name newtype's unwrapper `unCoprimes`.